integration_tests/src/main/python/data_gen.py

```diff
@@ -826,6 +826,19 @@ def gen_scalars_for_sql(data_gen, count, seed=0, force_no_nulls=False):
            decimal_gen_default, decimal_gen_scale_precision, decimal_gen_same_scale_precision,
            decimal_gen_64bit]
 
+common_cudf_types_gen = all_basic_gens[:-1]
+
+# Pyarrow will complain the error as below if the timestamp is out of range for both CPU and GPU,
+# so narrow down the time range to avoid exceptions causing test failures.
+#
+#     "pyarrow.lib.ArrowInvalid: Casting from timestamp[us, tz=UTC] to timestamp[ns]
+#      would result in out of bounds timestamp: 51496791452587000"
+#
+# This issue has been fixed in pyarrow by the PR https://github.com/apache/arrow/pull/7169
+# However it still requires PySpark to specify the new argument "timestamp_as_object".
+arrow_common_gen = common_cudf_types_gen[:-1] + [TimestampGen(
```

Comment:
Same slicing problem here.  We should build up from common types rather than assume we can slice out what we need.

Reply:
Updated accordingly.